workers/go/workflows/utils.go

```diff
@@ -0,0 +1,29 @@
+package workflows
+
+import (
+	"fmt"
+	"go.temporal.io/sdk/workflow"
+)
+
+// RunConcurrently runs the given functions concurrently, waiting for all to complete.
+// Returns early if any of the functions return an error.
+func RunConcurrently(ctx workflow.Context, funcs ...func(workflow.Context) error) error {
+	completed := 0
+	var innerErr error
+	for _, f := range funcs {
+		f := f
+		workflow.Go(ctx, func(ctx workflow.Context) {
+			innerErr = f(ctx)
+			completed++
+		})
+	}
+
+	awaitErr := workflow.Await(ctx, func() bool { return completed == len(funcs) || innerErr != nil })
```

Comment:
FWIW, there is a https://pkg.go.dev/go.temporal.io/sdk/workflow#WaitGroup for this use case, though what you have here is fine too

Reply:
Figured it was unnecessary since we aren't really running in parallel.